templates/federated_reporting/mrproper.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+"""
+mrproper.py - a script to remove hosts which have migrated to
+other feeder hubs. To be run on Federated Reporting superhub
+after each import of feeder data.
+
+First, to setup, enable mrproper by setting a class in augments (def.json).
+This enables policy in cfe_internal/enterprise/federation/federation.cf
+
+```json
+{
+  "classes": {
+    "cfengine_mp_fr_enable_mrproper": [ "any::" ]
+  }
+}
+```
+
+After the policy has run on superhub and feeders, run this script
+to setup mrproper role and account on all feeders and superhubs with
+proper RBAC settings for normal operation.
+You will be prompted for superhub admin credentials and then
+admin credentials on each feeder.
+"""
+
+import os
+import platform
+import string
+import random
+import sys
+from getpass import getpass
+from nova_api import NovaApi
+from cfsecret import read_secret, write_secret
+
+if not os.geteuid() == 0:
+    sys.exit("\n{} must be run as root".format(os.path.basename(__file__)))
+
+# allow environment override of cfengine WORKDIR
+WORKDIR = os.environ.get("WORKDIR", "/var/cfengine")
+
+# Primary dir in which to place various needed files
+MRPROPER_DIR = "/opt/cfengine/federation/cftransport/mrproper"
+
+# collect cert files from /var/cfengine/httpd/ssl/certs on
+# superhub and feeders and cat all together into hubs.cert
+CERT_PATH = MRPROPER_DIR + "/hubs.cert"
```

Comment:
This (and the other path below) can use `os.path.join()`.

Reply:
ack